cmake/SourceList.cmake

```diff
@@ -16,11 +8,36 @@ function(add_files)
         endif()
     endif()
 
+    get_target_property(${tgt}_FILES ${tgt} SOURCES)
+
     foreach(FILE IN LISTS PARAM_FILES)
-        target_sources(openttd_lib PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/${FILE})
+        if(${tgt}_FILES MATCHES "/${FILE}(;|$)")
```

Comment:
Sorry, should have noticed this earlier: we don't mention here our intention of this. Would be good to add something along the lines of not all IDEs being happy with duplicated filenames. Just so we know in 5 years why we added this :D